build-ps/percona-server-5.7_builder.sh

```diff
@@ -0,0 +1,825 @@
+        --get_sources       Source will be downloaded from github
+        --build_src_rpm     If it is 1 src rpm will be built
+        --build_source_deb  If it is 1 source deb package will be built
+        --build_rpm         If it is 1 rpm will be built
+        --build_deb         If it is 1 deb will be built
+        --build_tarball     If it is 1 tarball will be built
+        --install_deps      Install build dependencies(root previlages are required)
+        --branch            Branch for build
+        --repo              Repo for build
+        --yassl             build tarball with yassl
+        --perconaft_repo    PerconaFT repo
+        --perconaft_branch  Branch for PerconaFT
+        --tokubackup_repo   TokuBackup repo
+        --tokubackup_branch Btanch for TokuBackup
+        --rpm_release       RPM version( default = 1)
+        --deb_release       DEB version( default = 1)
+        --help) usage ;;
+Example $0 --builddir=/tmp/PS57 --get_sources=1 --build_src_rpm=1 --build_rpm=1
+EOF
+        exit 1
+}
+
+append_arg_to_args () {
+  args="$args "$(shell_quote_string "$1")
+}
+
+parse_arguments() {
+    pick_args=
+    if test "$1" = PICK-ARGS-FROM-ARGV
+    then
+        pick_args=1
+        shift
+    fi
+  
+    for arg do
+        val=$(echo "$arg" | sed -e 's;^--[^=]*=;;')
+        case "$arg" in
+            # these get passed explicitly to mysqld
+            --builddir=*) WORKDIR="$val" ;;
+            --build_src_rpm=*) SRPM="$val" ;;
+            --build_source_deb=*) SDEB="$val" ;;
+            --build_rpm=*) RPM="$val" ;;
+            --build_deb=*) DEB="$val" ;;
+            --get_sources=*) SOURCE="$val" ;;
+            --build_tarball=*) TARBALL="$val" ;;
+            --branch=*) BRANCH="$val" ;;
+            --repo=*) REPO="$val" ;;
+            --install_deps=*) INSTALL="$val" ;;
+            --yassl=*) YASSL="$val" ;;
+            --perconaft_branch=*) PERCONAFT_BRANCH="$val" ;;
+            --tokubackup_branch=*) 	TOKUBACKUP_BRANCH="$val" ;;
+            --perconaft_repo=*) PERCONAFT_REPO="$val" ;;
+            --tokubackup_repo=*) TOKUBACKUP_REPO="$val" ;;
+            --rpm_release=*) RPM_RELEASE="$val" ;;
+            --deb_release=*) DEB_RELEASE="$val" ;;
+            --help) usage ;;      
+            *)
+              if test -n "$pick_args"
+              then
+                  append_arg_to_args "$arg"
+              fi
+              ;;
+        esac
+    done
+}
+
+check_workdir(){
+    if [ "x$WORKDIR" = "x$CURDIR" ]
+    then
+        echo >&2 "Current directory cannot be used for building!"
+        exit 1
+    else
+        if ! test -d "$WORKDIR"
+        then
+            echo >&2 "$WORKDIR is not a directory."
+            exit 1
+        fi
+    fi
+    return
+}
+
+add_percona_yum_repo(){
+    if [ ! -f /etc/yum.repos.d/percona-dev.repo ]
+    then
+        cat >/etc/yum.repos.d/percona-dev.repo <<EOL
```

Comment:
https://jenkins.percona.com/yum-repo/percona-dev.repo with keys in the same dir

Reply:
done